### PR TITLE
Feature/doma 370/prettier on husky

### PR DESCRIPTION
**Pull request by AntonAL**
Brought back PR #332 

It takes changed code (Git stage) and formats it unobtrusively in pre-commit hook.
This way, all lines of code, affected by next commits will have consistent formatting.

Some Prettier rules are derived from ESLint

Set line width to 130 chars to avoid changing overhead
Excluded:
- Non JS-files `.json`, `.md`, `.yml`, `.yaml`.
- Everything automatically generated, like `schema.d.ts`, `migrations`, `.kmigrator` etc.
- Items, listed in ESLint `ignorePatterns`

## Playing

Check sources with Prettier:

```
yarn prettier --write .
```

Pass Prettier over all sources and rewrite:

```
yarn prettier --write .
```

### 👉 Will Prettier be passed over all sources?

Yes, because Prettier is now integrated in Linting rules and all sources should conform described formatting rules.

~~It seems to be No, because of following reasons:~~
~~- To avoid misunderstanding, like I affected 11000+ lines of code manually ;)~~
~~- To preserve authors of all existing lines of code~~
~~- Some lines, marked with `@ts-ignore` get's unwrapped by Prettier and Linter reports errors. Fixing all of these errors are out of scope of Prettier integration~~

So, a commit `SBERDOMA-370 passed Prettier over all sources` is for information and not intended to be merged. It will be deleted before merge.

~~### Why tests in Linter tests are failing~~

~~Because Prettier has unwrapped one line with `ts-ignore` to several lines in `apps/condo/domains/common/utils/excel.utils.ts`.~~

### Why Prettier and not only ESLint?

Not always attention is paid to indication of ESLint errors in IDE.
Since Prettier is automatic, no human mistakes will be made.

### Why formatting is not performed on CI level?

Formatting on CI level will make mess to the changes history, because formatting will be applied:
- either by changing pushed commit on remote (we will get different local and remote SHA's);
- or by creating a separate commit (will bloat changes history).

Locally installed Prettier acts on pre-commit phase locally, and we get instant feedback.

### Why `@typescript-eslint/indent` is disabled?

Because there is a clash of indentation rules between Prettier and ESLint in following case:

```
const { useObject, useObjects, useCreate, useUpdate, useDelete } = generateReactHooks<
    BillingAccount,
    BillingAccountUpdateInput,
    IBillingAccountFormState,
    IBillingAccountUIState,
    QueryAllBillingAccountsArgs
>(BillingAccountGQL, { convertToGQLInput, convertToUIState }
```

This code, formatted by Prettier, is handled as incorrect by ESLint.
I did't found how to control ESLint especially for angle-brackets.
Just turned it off, it's not critical, IMHO.
https://stackoverflow.com/a/58977894/235158

### OMG, "this", and "that" is now different

We always had tens of thousands lines of inconsistent code style ;)
Now it will be consistent.

### CODE REVIEW

File: apps/condo/domains/acquiring/schema/MultiPayment.js [hunk 1]
```diff
@@ -55,9 +55,8 @@ const access = require('@condo/domains/acquiring/access/MultiPayment')
 const get = require('lodash/get')
 const Big = require('big.js')
 
-
 const MultiPayment = new GQLListSchema('MultiPayment', {
-    schemaDoc: 'Information about resident\'s payment for single or multiple services/receipts',
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.js:60]: does it conflict with linter?

File: .huskyrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# This loads nvm.sh and sets the correct PATH before running hook
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+
+nvm use 14
```
Comment (pahaz) [inline .huskyrc:5]: I don't think that it's a good idea to add it for all.

Comment (pahaz) [reply .huskyrc:5]: it's better to add it inside your `~/.bashrc`

File: apps/condo/domains/acquiring/schema/MultiPayment.js (hunk 1 above)
Comment (AntonAL) [reply apps/condo/domains/acquiring/schema/MultiPayment.js:60]: No, you can inspect the code using `yarn lint --quiet`

File: .huskyrc (hunk 2 above)
Comment (AntonAL) [reply .huskyrc:5]: Discussed.
Removed it and created a documentation

**Comment by pahaz** [comment]
It's tooo old and no activity ...

**Pull request closed without merging**